File: app/src/main/java/com/remedios/eclassrecordteacher/classes/UserDao.java
package com.remedios.eclassrecordteacher.classes;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

@Dao
public interface UserDao {

    @Insert
    void insert (Users users);

    @Update
    void update(Users users);

    @Query("delete from Users where id=:id")
    void delete(int id);

    @Query("Select * from Users")
    void getAllUsers();



}

File: app/src/main/java/com/remedios/eclassrecordteacher/student/UserDao.kt
package com.remedios.eclassrecordteacher.student

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update
import androidx.sqlite.db.SupportSQLiteQuery


@Dao
interface UserDao {

    @Insert
    suspend fun addUser(user: User)

    @Query("SELECT * FROM user")
    suspend fun getAllUser():List<User>

    @Update
    suspend fun updateUser(user: User)

    @Delete
    suspend fun deleteUser(user: User)

//    @Query("SELECT * FROM user JOIN teacherUser ON user.id = teacherUser.id")
//    suspend fun getAllStudents():Map<User, List<TeacherUser>>
//    fun insertDataRawFormat(query: SupportSQLiteQuery): Boolean?

}